src/HTTPRequest.cpp

```diff
@@ -62,21 +63,36 @@ void HTTPRequest::get(const URL& url,
                       std::function<void(const std::string&)> onError,
                       const std::string& fileName)
 {
-    try
+    auto getAttempts {2};
+    std::string exceptionMessage;
+
+    // Try the request 'getAttempts' times
+    while (0 < getAttempts)
     {
-        auto req {GetRequest::builder(FactoryRequestWrapper<wrapperType>::create())};
-        req.url(url.url())
-            .appendHeader("Content-Type: application/json")
-            .appendHeader("Accept: application/json")
-            .appendHeader("Accept-Charset: utf-8")
-            .outputFile(fileName)
-            .execute();
+        try
+        {
+            auto req {GetRequest::builder(FactoryRequestWrapper<wrapperType>::create())};
+            req.url(url.url())
+                .appendHeader("Content-Type: application/json")
+                .appendHeader("Accept: application/json")
+                .appendHeader("Accept-Charset: utf-8")
+                .outputFile(fileName)
+                .execute();
 
-        onSuccess(req.response());
+            onSuccess(req.response());
+            break;
+        }
+        catch (const std::exception& ex)
+        {
+            getAttempts--;
+            exceptionMessage = ex.what();
```

Comment:
Improve (design): If the error message differs from one execution to the other, this will mask that.
I think we should capture all of them
```suggestion
            exceptionMessage += "'" + getAttempex.what() + "'. ";
```

Reply:
An alternative is to save them in a vector but that involves a lot of changes

Reply:
I find this request not very clean and/or scalable.

Also, the callback will receive a string of concatenated errors: complex to parse IMO (the caller should be responsible for how to log this error).

For N attempts, the resulting error would be:
```
"'Error 1 message'. 'Error 2 message'. ... 'Error N message'. "
```

Reply:
Done.

Reply:
[Here](https://github.com/wazuh/wazuh-http-request/blob/b36a0cc870f0d285bb1a3465df71756fd3760398/test/component/component_test.cpp#L108) is an example on how the message will be generated.